drivers/soundwire/intel.c

```diff
@@ -1977,17 +1981,23 @@ static const struct dev_pm_ops intel_pm = {
 	SET_RUNTIME_PM_OPS(intel_suspend_runtime, intel_resume_runtime, NULL)
 };
 
-static struct platform_driver sdw_intel_drv = {
-	.probe = intel_master_probe,
-	.remove = intel_master_remove,
+static const struct auxiliary_device_id intel_link_id_table[] = {
+	{ .name = "soundwire_intel.link" },
+	{},
+};
+MODULE_DEVICE_TABLE(auxiliary, intel_link_id_table);
+
+static struct auxiliary_driver sdw_intel_drv = {
+	.probe = intel_link_probe,
+	.remove = intel_link_remove,
+	.shutdown = intel_link_shutdown,
 	.driver = {
 		.name = "intel-sdw",
```

Comment:
@plbossart auxiliary_driver_register() sets the name to be the modname. I think this is not needed here.

Reply:
indeed, I didn't recall that detail. thanks!